janusgraph-test/src/main/java/org/janusgraph/graphdb/JanusGraphIndexTest.java

```diff
@@ -988,6 +992,57 @@ public void testRawQueries() {
         assertCount(numV / strs.length * 2, graph.indexQuery(PINDEX, "text:ducks").properties());
     }
 
+    /**
+     * Tests query parameters with raw indexQuery
+     */
+    @Test
+    public void testRawQueriesWithParameters() throws Exception {
+        if (!supportsLuceneStyleQueries()) return;
+        Parameter asc_sort_p = null;
+        Parameter desc_sort_p = null;
+
+        // ElasticSearch and Solr have different formats for sort parameters
+        if (this.getClass().getSimpleName().toLowerCase().contains("elasticsearch")) {
+            Map<String,String> sortAsc = new HashMap<String,String>();
+            sortAsc.put("_score", "asc");
+            asc_sort_p = new Parameter("sort",Arrays.asList(sortAsc));
+            Map<String,String> sortDesc = new HashMap<String,String>();
+            sortDesc.put("_score", "desc");
+            desc_sort_p = new Parameter("sort",Arrays.asList(sortDesc));
+        } else if (this.getClass().getSimpleName().toLowerCase().contains("solr")) {
+            asc_sort_p = new Parameter("sort",new String[]{"score asc"});
+            desc_sort_p = new Parameter("sort",new String[]{"score desc"});
+        } else {
+            return;  // Ignore for !Solr or !ElasticSearch
+        }
+        
+        final PropertyKey field1Key = mgmt.makePropertyKey("field1").dataType(String.class).make();
+        mgmt.buildIndex("store1", Vertex.class).addKey(field1Key).buildMixedIndex(INDEX);
+        mgmt.commit();
+
+        JanusGraphVertex v1 = tx.addVertex();
+        JanusGraphVertex v2 = tx.addVertex();
+        JanusGraphVertex v3 = tx.addVertex();
+
+        v1.property("field1", "Hello Hello Hello Hello Hello Hello Hello Hello");
+        v2.property("field1", "Hello blue and yellow meet green");
+        v3.property("field1", "Hello");
+
+        tx.commit();
+
+        Thread.sleep(5000);
```

Comment:
The sleep is to give the index and graph time to commit since the commit() is not guaranteed to be complete upon return.  This pattern was copied from "testIndexQueryWithScore()" and used other places in the unit test.  I've opened a separate issue #429 to replace the use of Thread.sleep with tx().addTransactionListener() as documented in my response on issue #329.